src/cpp/src/lm_encoding.cpp

```diff
@@ -0,0 +1,224 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <numeric>
+#include <random>
+#include <regex>
+#include <vector>
+
+#include "lm_encoding.hpp"
+#include "openvino/genai/perf_metrics.hpp"
+
+
+namespace ov {
+namespace genai {
+
+// void reset_all_inputs_to_empty_tensors(ov::InferRequest& request) {
+//     request.set_tensor("input_ids", ov::Tensor(ov::element::i64, {0, 0}));
+//     request.set_tensor("beam_idx", ov::Tensor(ov::element::i32, {0}));
+//     if (request.get_compiled_model().inputs().size() == 4)
+//         request.set_tensor("position_ids", ov::Tensor(ov::element::i64, {0, 0}));
+// }
+
+
+EncodedResults get_lm_encoded_results(
+    ov::InferRequest& m_model_runner,
+    const ov::Tensor& input_ids,
+    ov::Tensor attention_mask,
+    const std::shared_ptr<StreamerBase>& streamer_ptr,
+    Sampler& sampler,
+    std::vector<SequenceGroup::Ptr> sequence_groups,
+    std::optional<ov::Tensor> position_ids,
+    std::optional<ov::InferRequest> m_embedding,
+    std::optional<size_t> hidden_size,
+    std::optional<float> scale_emb
```

Comment:
can we merged this scaling as post-processing for embedding model?
In this case we will not have this parameter.

It can be done as separate PR as prerequisite for this PR.

In this case we can connect embedding and LLM models via ov::RemoteTensor, which is required for optimal performance on GPU

Reply:
done in PR https://github.com/openvinotoolkit/openvino.genai/pull/1066